### PR TITLE
core(runner): save lhr on -A

### PR DESCRIPTION
**Pull request by connorjclark**

### CODE REVIEW

File: lighthouse-core/runner.js [hunk 1]
```diff
@@ -156,6 +156,12 @@ class Runner {
       // LHR has now been localized.
       const lhr = /** @type {LH.Result} */ (i18nLhr);
 
+      // Save lhr to ./latest-run, but only if -A is used.
+      if (settings.auditMode) {
+        const path = Runner._getDataSavePath(settings);
```
Comment (connorjclark) [inline lighthouse-core/runner.js:161]: I noticed an lhr file will linger if you did a `-G`  ... I nearly wrote some code to delete the latest run folder before running, but that seems dangerous.

Comment (patrickhulce) [reply lighthouse-core/runner.js:161]: > I nearly wrote some code to delete the latest run folder before running, but that seems dangerous.

thank goodness you didn't, that would have been horrifying to me and the important examples I keep in that folder due to its gitignore properties 😅

**Review by adamraine — COMMENTED** [review]
Looks good, can you update the readme with new description for `-A`?

Screen Shot 2020-10-02 at 3 00 35 PM

File: lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
+{
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json:1]: This doesn't bother me, and I'd prefer to avoid the churn.

**Comment by connorjclark** [comment]
> Looks good, can you update the readme with new description for -A?

I actually think we should avoid mentioning in the readme, because we don't want people to get the idea that this is how you get a LHR. I think it'd be more confusing than anything. this feature is really only meant for contributors imo.

**Review by adamraine — APPROVED** [review]
LGTM

**Comment by brendankenny** [comment]
> because we don't want people to get the idea that this is how you get a LHR. I think it'd be more confusing than anything. this feature is really only meant for contributors imo.

Contributors read readmes, too :) It seems as worth documenting as anything else in that screenshot.

File: lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json:1]: Looks like this breaks our travis assertion that no source files change by running the tests (something we should add to github assertion suite 👍 ). 

Actually looking at the code now, I'm a little confused why are these checked in at all. It doesn't look like they're normalized, and we never check equality (or even their existence?). Seems like an accidental add with `git -A`.

[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json:1]: > something we should add to github assertion suite 👍

soooo we already do this and now I'm confused why github passes but travis doesn't...

[the remark this answers is not in the document]
Comment (brendankenny) [reply lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json:1]: > figured it out we don't run it after the unit tests!

😢 😢 

Should we fix that and revert these lhrs (unless there was a non-apparent reason to add them)? Future tests that might need an lhr with these artifacts can always generate them (and normalize them, etc) when needed.

[the remark this answers is not in the document]
Comment (brendankenny) [reply lighthouse-core/test/fixtures/artifacts/alphabet-artifacts/lhr.json:1]: lol you're already on it

**Pull request merged**